clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1287,6 +1287,16 @@ public void testCommitAsyncNegativeOffset() {
         assertTrue(mockOffsetCommitCallback.exception instanceof IllegalArgumentException);
     }
 
+    @Test(expected = CommitFailedException.class)
+    public void testCommitOffsetSyncCallbackWithoutNonRetriableException() {
```

Comment:
Sorry, jumped the gun on my +1. I don't think this is quite right. We need to hit the case when the request times out, right?